### PR TITLE
Added support for Julia language

### LINKED ISSUE #34: Need Support for Julia Lang
As a code comment extractor, We are planning to increase our reach to preferred programming languages world wide. We need support for Julia.

For getting started you can refer to [CONTRIBUTING.md](https://github.com/fossology/Nirjas/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by soham4abc**
Fixes #34 

Please can anyone help me or guide me to let me know where my tests are failing?

@GMishx @Kaushl2208

### CODE REVIEW

**Comment by Kaushl2208** [comment]
> Fixes #34
> 
> Please can anyone help me or guide me to let me know where my tests are failing?
> 
> @GMishx @Kaushl2208

You need to change the testfile:
1. The current test file contains the unicode characters.
2. Run Nirjas on the same testfile and analyse the output and share it here as well(the output).

**Comment by soham4abc** [comment]
```
PS C:\Users\SOHAM\Desktop\misc\tempnirjas> nirjas .\test.jl
{
    "metadata": {
        "filename": ".\\test.jl",
        "lang": "Julia",
        "total_lines": 75,
        "total_lines_of_comments": 8,
        "blank_lines": 16,
        "sloc": 51
    },
    "single_line_comment": [
        {
            "line_number": 5,
            "comment": "get the category, don't process some files"
        },
        {
            "line_number": 15,
            "comment": "process the  lines"
        },
        {
            "line_number": 23,
            "comment": "handle subcategories"
        },
        {
            "line_number": 27,
            "comment": "collect repo info"
        },
        {
            "line_number": 56,
            "comment": "-----------------------------------------------------------------"
        },
        {
            "line_number": 61,
            "comment": "loop over all markdown files in the root directory, appending the records to the list"
        },
        {
            "line_number": 67,
            "comment": "save a csv file"
        },
        {
            "line_number": 75,
            "comment": "module"
        }
    ],
    "cont_single_line_comment": [],
    "multi_line_comment": []
}
PS C:\Users\SOHAM\Desktop\misc\tempnirjas>
```

**Comment by soham4abc** [comment]
This is what I get when I am running the same test file using nirjas. What I need to change? @Kaushl2208 

> ```
> PS C:\Users\SOHAM\Desktop\misc\tempnirjas> nirjas .\test.jl
> {
>     "metadata": {
>         "filename": ".\\test.jl",
>         "lang": "Julia",
>         "total_lines": 75,
>         "total_lines_of_comments": 8,
>         "blank_lines": 16,
>         "sloc": 51
>     },
>     "single_line_comment": [
>         {
>             "line_number": 5,
>             "comment": "get the category, don't process some files"
>         },
>         {
>             "line_number": 15,
>             "comment": "process the  lines"
>         },
>         {
>             "line_number": 23,
>             "comment": "handle subcategories"
>         },
>         {
>             "line_number": 27,
>             "comment": "collect repo info"
>         },
>         {
>             "line_number": 56,
>             "comment": "-----------------------------------------------------------------"
>         },
>         {
>             "line_number": 61,
>             "comment": "loop over all markdown files in the root directory, appending the records to the list"
>         },
>         {
>             "line_number": 67,
>             "comment": "save a csv file"
>         },
>         {
>             "line_number": 75,
>             "comment": "module"
>         }
>     ],
>     "cont_single_line_comment": [],
>     "multi_line_comment": []
> }
> PS C:\Users\SOHAM\Desktop\misc\tempnirjas>
> ```

File: nirjas/languages/tests/test_julia.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        expected = julia.juliaExtractor(self.testfile).get_dict()
+        comment_single = readSingleLine(self.testfile, regex)
+        comment_multi_single = readMultiLineSame(self.testfile, syntax_single)
+        comment_multi_double = readMultiLineSame(self.testfile, syntax_double)
+        comment_contSingleline = contSingleLines(comment_single)
+        file = self.testfile.split("/")
+        output = {
+            "metadata": {
+                "filename": file[-1],
+                "lang": "julia",
```
Comment (Kaushl2208) [inline nirjas/languages/tests/test_julia.py:69]:
```suggestion
                "lang": "Julia",
```

File: nirjas/languages/julia.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Copyright (C) 2020  Ayush Bhardwaj (classicayush@gmail.com),
+Kaushlendra Pratap (kaushlendrapratap.9837@gmail.com)
```
Comment (Kaushl2208) [inline nirjas/languages/julia.py:5]: You should change the copyright to your name.

**Comment by soham4abc** [comment]
All Checks passed @Kaushl2208  PTAL

File: testScript.py [hunk 3]
```diff
@@ -55,7 +55,8 @@ def download_files(cwd):
         "https://raw.githubusercontent.com/airbnb/react-with-styles/2532394bb866aaade4dc750ee94c0ff213d9b6de/src/withStyles.jsx",
         "https://raw.githubusercontent.com/flutter/plugins/e61e9d45bcaadc3e409d529d30735cb4db75c5c5/packages/android_alarm_manager/lib/android_alarm_manager.dart",
         "https://raw.githubusercontent.com/microsoft/TypeScript/c33a14d66d0a452673ce77256e178bf84e875d2b/tests/cases/user/formik/index.tsx",
-        "https://raw.githubusercontent.com/twbs/bootstrap/9488978fb55286ba83e8193a871d1ff9815045b9/scss/_reboot.scss"
+        "https://raw.githubusercontent.com/twbs/bootstrap/9488978fb55286ba83e8193a871d1ff9815045b9/scss/_reboot.scss",
+        "https://raw.githubusercontent.com/svaksha/Julia.jl/master/src/scrape.jl"
```
Comment (Kaushl2208) [inline testScript.py:59]: Change master ==> to a persistent commit id, As the changes in the master will usually get lost.

**Review by Kaushl2208 — CHANGES_REQUESTED** [review]

**Comment by soham4abc** [comment]
All requested changes done PTAL @Kaushl2208

**Comment by soham4abc** [comment]
Any more changes required @GMishx  @Kaushl2208 ??

**Review by Kaushl2208 — APPROVED** [review]
Changes LGTM, Needs test!

**Comment by soham4abc** [comment]
> Changes LGTM, Needs test!

Nice... Let me know if any more changes are required!

**Comment by soham4abc** [comment]
Any changes required? @Kaushl2208

**Comment by soham4abc** [comment]
> Any changes required? @Kaushl2208
@Kaushl2208  @GMishx ??

**Comment by Kaushl2208** [comment]
Hey @soham4abc , We will let you know once the testing has been completed for the PR.

Thankyou for your patience. :)

**Comment by Kaushl2208** [comment]
Hey  @soham4abc , I have tested it. Thankyou for your contributions.


CC: @GMishx

**Comment by soham4abc** [comment]
> Hey @soham4abc , I have tested it. Thankyou for your contributions.
> 
> Can you Squash the commits into one before I merge it? CC: @GMishx

changes merged to one commit and rebased

**Pull request merged**